File: api-tester/src/defaults/java/com/revenuecat/apitester/java/revenuecatui/PaywallFooterViewAPI.java
package com.revenuecat.apitester.java.revenuecatui;

import android.content.Context;
import android.util.AttributeSet;

import com.revenuecat.purchases.Offering;
import com.revenuecat.purchases.ui.revenuecatui.PaywallListener;
import com.revenuecat.purchases.ui.revenuecatui.fonts.FontProvider;
import com.revenuecat.purchases.ui.revenuecatui.views.OriginalTemplatePaywallFooterView;
import com.revenuecat.purchases.ui.revenuecatui.views.PaywallFooterView;

@SuppressWarnings({"unused"})
final class PaywallFooterViewAPI {

    static void checkConstructors(Context context,
                                  AttributeSet attrs,
                                  int defStyleAttr,
                                  Offering offering,
                                  PaywallListener listener,
                                  FontProvider fontProvider,
                                  Boolean condensed) {
        PaywallFooterView deprecatedFooterView = new PaywallFooterView(context);
        PaywallFooterView deprecatedFooterView2 = new PaywallFooterView(context, attrs);
        PaywallFooterView deprecatedFooterView3 = new PaywallFooterView(context, attrs, defStyleAttr);
        PaywallFooterView deprecatedFooterView4 = new PaywallFooterView(context, offering);
        PaywallFooterView deprecatedFooterView5 = new PaywallFooterView(context, offering, listener);
        PaywallFooterView deprecatedFooterView6 = new PaywallFooterView(context, offering, listener, fontProvider);
        PaywallFooterView deprecatedFooterView7 = new PaywallFooterView(context, offering, listener, fontProvider, condensed);
        PaywallFooterView deprecatedFooterView8 = new PaywallFooterView(context, offering, listener, fontProvider, condensed, () -> null);
        OriginalTemplatePaywallFooterView footerView = new OriginalTemplatePaywallFooterView(context);
        OriginalTemplatePaywallFooterView footerView2 = new OriginalTemplatePaywallFooterView(context, attrs);
        OriginalTemplatePaywallFooterView footerView3 = new OriginalTemplatePaywallFooterView(context, attrs, defStyleAttr);
        OriginalTemplatePaywallFooterView footerView4 = new OriginalTemplatePaywallFooterView(context, offering);
        OriginalTemplatePaywallFooterView footerView5 = new OriginalTemplatePaywallFooterView(context, offering, listener);
        OriginalTemplatePaywallFooterView footerView6 = new OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider);
        OriginalTemplatePaywallFooterView footerView7 = new OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider, condensed);
        OriginalTemplatePaywallFooterView footerView8 = new OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider, condensed, () -> null);
    }

    static void checkDeprecatedMethods(PaywallFooterView footerView, PaywallListener listener) {
        footerView.setPaywallListener(null);
        footerView.setPaywallListener(listener);
        footerView.setDismissHandler(null);
        footerView.setDismissHandler(() -> null);
        footerView.setOfferingId(null);
        footerView.setOfferingId("offeringId");
    }

    static void checkMethods(OriginalTemplatePaywallFooterView footerView, PaywallListener listener) {
        footerView.setPaywallListener(null);
        footerView.setPaywallListener(listener);
        footerView.setDismissHandler(null);
        footerView.setDismissHandler(() -> null);
        footerView.setOfferingId(null);
        footerView.setOfferingId("offeringId");
    }
}

File: api-tester/src/defaults/kotlin/com/revenuecat/apitester/kotlin/revenuecatui/PaywallFooterViewAPI.kt
package com.revenuecat.apitester.kotlin.revenuecatui

import android.content.Context
import android.util.AttributeSet
import android.widget.FrameLayout
import com.revenuecat.purchases.Offering
import com.revenuecat.purchases.ui.revenuecatui.PaywallListener
import com.revenuecat.purchases.ui.revenuecatui.fonts.FontProvider
import com.revenuecat.purchases.ui.revenuecatui.views.OriginalTemplatePaywallFooterView

@Suppress("unused", "UNUSED_VARIABLE")
private class PaywallFooterViewAPI {

    fun checkType(context: Context) {
        val paywallFooterView: FrameLayout = OriginalTemplatePaywallFooterView(context)
    }

    @Suppress("LongParameterList")
    fun checkConstructors(
        context: Context,
        attrs: AttributeSet?,
        defStyleAttr: Int,
        offering: Offering,
        listener: PaywallListener,
        fontProvider: FontProvider,
        condensed: Boolean,
        dismissHandler: () -> Unit,
    ) {
        OriginalTemplatePaywallFooterView(context)
        OriginalTemplatePaywallFooterView(context, attrs)
        OriginalTemplatePaywallFooterView(context, attrs, defStyleAttr)
        OriginalTemplatePaywallFooterView(context, offering)
        OriginalTemplatePaywallFooterView(context, offering, listener)
        OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider)
        OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider, condensed)
        OriginalTemplatePaywallFooterView(context, offering, listener, fontProvider, condensed, dismissHandler)
    }

    fun checkMethods(
        paywallFooterView: OriginalTemplatePaywallFooterView,
        paywallListener: PaywallListener,
        dismissHandler: () -> Unit,
    ) {
        paywallFooterView.setPaywallListener(null)
        paywallFooterView.setPaywallListener(paywallListener)
        paywallFooterView.setDismissHandler(null)
        paywallFooterView.setDismissHandler(dismissHandler)
        paywallFooterView.setOfferingId(null)
        paywallFooterView.setOfferingId("offering_id")
    }
}
